build.gradle.kts

```diff
@@ -300,4 +319,47 @@ fun PublicationContainer.setDefaultPomFields() {
       }
     }
   }
+}
+
+tasks.register("publishIfNeeded") {
+  val eventName = System.getenv("GITHUB_EVENT_NAME")
+  val ref = System.getenv("GITHUB_REF")
+  project.logger.log(LogLevel.LIFECYCLE, "publishIfNeeded eventName=$eventName ref=$ref")
+
+  if (eventName == "push" && ref == "ref/heads/master") {
+    project.logger.log(LogLevel.LIFECYCLE, "Deploying snapshot to OSS...")
+    dependsOn(subprojects.flatMap {
+      tasks.matching {
+        if (it.name == "apollo-gradle-plugin") {
+          it.name in arrayOf("publishDefaultPublicationToOjoRepository",
+              "publishApolloGradlePluginPluginMarkerMavenPublicationToOjoRepository")
+        } else {
+          it.name == "publishAllPublicationsToOjoRepository"
+        }
```

Comment:
@martinbonnin 
How can `it.name == "apollo-gradle-plugin"` be true and `it.name in arrayOf("publishDefaultPublicationToOjoRepository",
              "publishApolloGradlePluginPluginMarkerMavenPublicationToOjoRepository")` be true at the same time?

The content of that `if` branch looks incorrect as to what would be a intent that makes sense.

Reply:
Yup, that's a bug, the first `it` should be the project, the second `it` should be the task...
We have a very complicated bintray setup right now because we have 1 bintray package for each artifact, which doesn't really scale. I'm planning to streamling this for 2.0 and this code should actually disappear